### PR TITLE
CI: Add Xcode builds to GitHub Actions

### PR DESCRIPTION
**Pull request by ccawley2011**

### CODE REVIEW

File: devtools/create_project/xcode.cpp [hunk 1]
```diff
@@ -588,6 +600,9 @@ void XcodeProvider::setupFrameworksBuildPhase(const BuildSetup &setup) {
 	if (CONTAINS_DEFINE(setup.defines, "USE_ZLIB")) {
 		frameworks_iOS.push_back("libz.tbd");
 	}
+	if (CONTAINS_DEFINE(setup.defines, "USE_DISCORD")) {
+		frameworks_iOS.push_back("libdiscord-rpc.a");
```
Comment (criezy) [inline devtools/create_project/xcode.cpp:604]: I don't think it makes sense to use the discord feature with iOS. The way it is implemented currently means it is tied to the SDL backends. Furthermore I think it requires Discord and ScummVM running at the same time, which is not possible on iOS.

**Review by criezy — COMMENTED** [review]
Lately the Travis macOS build seems to have been working properly, but we know that it can have timeout issues. So switching to GitHub actions might be a good idea.

It's also nice that we can add an iOS build to the GitHub Actions and that it seems to work.

File: devtools/create_project/xcode.cpp (hunk 1 above)
Comment (SupSuper) [reply devtools/create_project/xcode.cpp:604]: Correct, discord-rpc only does anything on desktop platforms. There's no game activity on mobile Discord.

**Comment by SupSuper** [comment]
Good idea, Github Actions seems more stable and flexible than Travis. Might also be worth moving Linux over in the future.

**Comment by sev-** [comment]
Let's give it a try. Merging

**Pull request merged**